### PR TITLE
Performance tweaks

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

* Packets getting cloned in banking_stage is bad for perf.
* Packets getting received into a single buffer causes many packets buffers to be created and those buffers propagate poorly through the backend because of how they are batched.
* Sigverify coalescing of huge batches causes poor banking_stage parallelism.

#### Summary of Changes

* Use Rc to take references to Packets buffers to prevent copies.
* Coalesce incoming packets better to create larger batches for sigverify/banking and reduce allocator work and memory usage.
* Split up batches going into banking from sigverify to help banking_stage parallelism.

Fixes #

### CODE REVIEW

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -67,7 +70,7 @@ impl BankingStage {
             poh_recorder,
             verified_receiver,
             verified_vote_receiver,
-            cmp::min(2, Self::num_threads()),
+            cmp::max(2, Self::num_threads()),
```
[the remark this answers is not in the document]
Comment (sakridge) [reply core/src/banking_stage.rs]:
```
min(2, 16) = 2
max(2, 16) = 16
min(2, 1) = 1
max(2, 1) = 2
```

[the remark this answers is not in the document]
Comment (sakridge) [reply core/src/banking_stage.rs]: You want a minimum of 2 threads. So you want the max of 2 and the CPU core count.

**Comment by sakridge** [comment]
@pgarg66 I see about a 30-40% increase in sustained and max in a single machine case:
peak goes from 55k => 81k and average goes from 35k => 52k.

File: core/src/banking_stage.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pgarg66) [reply core/src/banking_stage.rs]: 👍

**Comment by pgarg66** [comment]
> @pgarg66 I see about a 30-40% increase in sustained and max in a single machine case:
> peak goes from 55k => 81k and average goes from 35k => 52k.

That's awesome! I can try it on the multinode testnet once it's merged.

**Pull request merged**